### DISCUSSION TITLE
Profile Update Error

### DISCUSSION THREAD

**Discussion by ruchinksinha** [Sunbird-Ed/Community#459] [Bugs] [answered]
Profile Update is giving an error - HTTP 400 Bad Request. Below is the request/response

**URL:** https://<domain_name>/learner/user/v3/update

**Request:**
{"params":{},"request":{"userId":"ed5b6945-c92f-4cc5-9c16-d009558e653b","profileUserTypes":[{"type":"administrator"}]}}

**Response:**
{
    "id": "api.user.update",
    "ver": "v3",
    "ts": "2023-04-13 18:21:38:971+0000",
    "params": {
        "resmsgid": "64172d5d-dbfb-56dd-a347-bedaa58b9af8",
        "msgid": "64172d5d-dbfb-56dd-a347-bedaa58b9af8",
        "err": "UOS_USRUPDnull",
        "status": "FAILED",
        "errmsg": "userType config is empty for the statecode default"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}

Please help in resolving this issue.

**Comment by ruchinksinha** [comment]
@swayangjit @luckynvdu - Please help in resolving this issue.

**Reply by swayangjit** [reply]
@ruchinksinha is it failing for a specific state ?

**Reply by ruchinksinha** [reply]
I have just installed the Sunbird Ed 5.1, executed all the pipelines are per the documentation and registered a new users. Now I am trying to login with that user and getting this error while profile update.

**Reply by swayangjit** [reply]
@ruchinksinha  Can you check if the following form is available in your instance or not
`curl --location 'https://<domain_name>/api/data/v1/form/read' \
--header 'Content-Type: application/json' \
--data '{
  "request": {
    "type": "profileconfig_v2",
    "subType": "default",
    "action": "get"
  }
}'`

**Reply by ruchinksinha** [reply]
@swayangjit 
This curl command didn't return any data. So I executed the same on https://staging.sunbirded.org/api/data/v1/form/read, and created the same in my environment. Now, the output of the above curl command is below: 

{"id":"api.form.read","params":{"resmsgid":"5763255f-093a-4ab4-8b43-d0983e41702f","msgid":"6abe4acd-e959-4bea-97c5-7a1857d7bf6a","status":"successful"},"responseCode":"OK","result":{"form":{"type":"profileconfig_v2","subtype":"*","action":"get","component":"*","framework":"*","data":{"templateName":"profileconfig_v2","action":"get","fields":[{"code":"name","type":"input","templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"Name"}},"hidden":true,"placeHolder":"Enter Name","multiple":false},"validations":[{"type":"required"}]},{"code":"persona","type":"nested_select","templateOptions":{"hidden":true,"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"Role"}},"placeHolder":"Select Role","multiple":false,"themeType":"material","dataSrc":{"marker":"SUPPORTED_PERSONA_LIST"}},"validations":[{"type":"required"}],"children":{"administrator":[{"code":"state","type":"select","templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"State"}},"placeHolder":"Select State","multiple":false,"dataSrc":{"marker":"STATE_LOCATION_LIST","params":{"useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"district","type":"select","context":"state","default":null,"templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"District"}},"placeHolder":"Select District","multiple":false,"dataSrc":{"marker":"LOCATION_LIST","params":{"id":"district","useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"block","type":"select","context":"district","default":null,"templateOptions":{"label":"Block","placeHolder":"Select Block","multiple":false,"dataSrc":{"marker":"LOCATION_LIST","params":{"id":"block","useCase":"SIGNEDIN"}}},"validations":[]},{"code":"cluster","type":"select","context":"block","default":null,"templateOptions":{"label":"Cluster","placeHolder":"Select Cluster","multiple":false,"dataSrc":{"marker":"LOCATION_LIST","params":{"id":"cluster","useCase":"SIGNEDIN"}}}},{"code":"school","type":"select","context":"cluster","default":null,"templateOptions":{"label":"School","placeHolder":"Select School","multiple":false,"dataSrc":{"marker":"LOCATION_LIST","params":{"id":"school","useCase":"SIGNEDIN"}}}}],"teacher":[{"code":"state","type":"select","templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"State"}},"placeHolder":"Select State","multiple":false,"themeType":"material","dataSrc":{"marker":"STATE_LOCATION_LIST","params":{"useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"district","type":"select","context":"state","default":null,"templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"District"}},"placeHolder":"Select District","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"district","useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"block","type":"select","context":"district","default":null,"templateOptions":{"label":"Block","placeHolder":"Select Block","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"block","useCase":"SIGNEDIN"}}},"validations":[]},{"code":"cluster","type":"select","context":"block","default":null,"templateOptions":{"label":"Cluster","placeHolder":"Select Cluster","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"cluster","useCase":"SIGNEDIN"}}}},{"code":"school","type":"select","context":"cluster","default":null,"templateOptions":{"label":"School","placeHolder":"Select School","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"school","useCase":"SIGNEDIN"}}}}],"student":[{"code":"state","type":"select","templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"State"}},"placeHolder":"Select State","multiple":false,"themeType":"material","dataSrc":{"marker":"STATE_LOCATION_LIST","params":{"useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"district","type":"select","context":"state","default":null,"templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"District"}},"placeHolder":"Select District","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"district","useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"block","type":"select","context":"district","default":null,"templateOptions":{"label":"Block","placeHolder":"Select Block","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"block","useCase":"SIGNEDIN"}}},"validations":[]},{"code":"cluster","type":"select","context":"block","default":null,"templateOptions":{"label":"Cluster","placeHolder":"Select Cluster","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"cluster","useCase":"SIGNEDIN"}}}},{"code":"school","type":"select","context":"cluster","default":null,"templateOptions":{"label":"School","placeHolder":"Select School","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"school","useCase":"SIGNEDIN"}}}}],"other":[{"code":"state","type":"select","templateOptions":{"placeHolder":"Select State","multiple":false,"themeType":"material","dataSrc":{"marker":"STATE_LOCATION_LIST","params":{"useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"district","type":"select","context":"state","default":null,"templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"District"}},"placeHolder":"Select District","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"district","useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"block","type":"select","context":"district","default":null,"templateOptions":{"label":"Block","placeHolder":"Select Block","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"block","useCase":"SIGNEDIN"}}},"validations":[]},{"code":"cluster","type":"select","context":"block","default":null,"templateOptions":{"label":"Cluster","placeHolder":"Select Cluster","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"cluster","useCase":"SIGNEDIN"}}}},{"code":"school","type":"select","context":"cluster","default":null,"templateOptions":{"label":"School","placeHolder":"Select School","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"school","useCase":"SIGNEDIN"}}}}],"parent":[{"code":"state","type":"select","templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"State"}},"placeHolder":"Select State","multiple":false,"themeType":"material","dataSrc":{"marker":"STATE_LOCATION_LIST","params":{"useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"district","type":"select","context":"state","default":null,"templateOptions":{"labelHtml":{"contents":"<span>$0&nbsp;<span class=\"required-asterisk\">*</span></span>","values":{"$0":"District"}},"placeHolder":"Select District","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"district","useCase":"SIGNEDIN_GUEST"}}},"validations":[{"type":"required"}]},{"code":"block","type":"select","context":"district","default":null,"templateOptions":{"label":"Block","placeHolder":"Select Block","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"block","useCase":"SIGNEDIN"}}},"validations":[]},{"code":"cluster","type":"select","context":"block","default":null,"templateOptions":{"label":"Cluster","placeHolder":"Select Cluster","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"cluster","useCase":"SIGNEDIN"}}}},{"code":"school","type":"select","context":"cluster","default":null,"templateOptions":{"label":"School","placeHolder":"Select School","multiple":false,"themeType":"material","dataSrc":{"marker":"LOCATION_LIST","params":{"id":"school","useCase":"SIGNEDIN"}}}}]}}]},"created_on":"2023-04-14T09:24:35.563Z","last_modified_on":null,"rootOrgId":"*"}},"ts":"2023-04-14T09:24:41.201Z","ver":"1.0"}



Do I need to restart any service after creating this form? I am still getting the error.
URL: https://<domain_name>/learner/user/v3/update
**Request:** 
{"params":{},"request":{"userId":"ed5b6945-c92f-4cc5-9c16-d009558e653b","profileUserTypes":[{"type":"administrator"}]}}

**Response:**
{
    "id": "api.user.update",
    "ver": "v3",
    "ts": "2023-04-14 09:26:21:018+0000",
    "params": {
        "resmsgid": "6426c0a7-5546-b3c6-1022-4796465b3b41",
        "msgid": "6426c0a7-5546-b3c6-1022-4796465b3b41",
        "err": "UOS_USRUPDnull",
        "status": "FAILED",
        "errmsg": "userType config is empty for the statecode default"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}

**Reply by swayangjit** [reply]
@ruchinksinha You can restart the learner service and verify

**Reply by ruchinksinha** [reply]
@swayangjit I tried restarting the learner pod as well as the learning tomcat on 10.0.1.7 but I am still getting the below error.

{
    "id": "api.user.update",
    "ver": "v3",
    "ts": "2023-04-14 09:42:22:809+0000",
    "params": {
        "resmsgid": "3c23743f-8e3a-8da6-0a8e-c0ea798c0492",
        "msgid": "3c23743f-8e3a-8da6-0a8e-c0ea798c0492",
        "err": "UOS_USRUPDnull",
        "status": "FAILED",
        "errmsg": "userType config is empty for the statecode default"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}

**Reply by chitranshu-keshav** [reply]
@swayangjit can you please get back on the above

**Reply by ruchinksinha** [reply]
@swayangjit Any additional troubleshoot steps can you suggest?

**Reply by swayangjit** [reply]
@ruchinksinha It seems to be a LERN build block issue. I have created a LERN [discussion thread](https://github.com/orgs/Sunbird-Lern/discussions/106) for the same. In the meantime, what are you trying to build, and which build blocks have you setup done till now?

**Reply by ruchinksinha** [reply]
@swayangjit 

I am trying to install the Sunbird-ED 5.1 Platform. I have followed the document https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0 for installation. All the suggested pipelines were successful.

Now I have created a login in the portal and when I try to login, I am getting the profile update error on the below form.

![image](https://user-images.githubusercontent.com/10114264/232728982-2c5ad90b-f0a5-4ad6-827e-7f2e84be4de6.png)

**Reply by swayangjit** [reply]
@ruchinksinha As suggested in the LERN discussion thread can you pls share the logs from learner pod?

**Reply by ruchinksinha** [reply]
@swayangjit shared the same.

**Comment by chitranshu-keshav** [comment]
@ruchinksinha The information provided in the thread is not sufficient to provide assistance on issue that you have reported. Can you please provide additional detail so that the community can help you out with resolving the issue. Some examples of additional details are below

1. Issue is being faced in which version of Sunbird ED?
2. Is this issue faced only with a specific content or all content of the content type?
3. Portal or mobile issue. If mobile, then it is being faced in all the mobile devices or a specific mobile device ?
4. Is it happening for a specific version of mobile OS ?
5. In which version of ED mobile app is this issue replicated?
6. Share the logs and the analysis done so far by your team and call out what specific help you need.

**Comment by ruchinksinha** [comment]
Hi,

Below are the details:

1) Version 5.1
2) All content on Portal and I haven't tried the mobile part yet

I am not sure what logs to share. Please let me know and I can share.

**Reply by swayangjit** [reply, marked as answer]
This [discussion thread](https://github.com/orgs/Sunbird-Lern/discussions/106#discussioncomment-5672073) resolves the issue
